### ISSUE TITLE
Attributes with utf8 chars gets converted to base64.

### ISSUE THREAD

**Issue by klausdk** [simplesamlphp/simplesamlphp-module-ldap#19]
This happens after this change : 

https://github.com/simplesamlphp/simplesamlphp-module-ldap/blob/2a3f12cb0909b7770ad3efa6b1299a176a0893c6/lib/Auth/Ldap.php#L861

Its a problem with first and last names containing danish characters like æøå.

**Comment by tvdijen** [comment]
You have an example for me, because I tested with `value = 'Knäckebrøt'` and the first half of the expression (mb_detect_encoding) comes out negative..

**Comment by klausdk** [comment]
"Møller" gets converted to base64. It also converts "Knäckebrøt" to base64
The first condition is false (`mb_detect_encoding($value) === false`),
but the second is true ( `preg_match('~[^\x20-\x7E\t\r\n]~', $value) > 0` )
The code was recently changed from an AND (&&) condition to an OR (||) condition.

**Comment by tvdijen** [comment]
Ah right, my bad.. I'm gonna loosen the preg_match to match extended ascii as well.. That should fix it

**Comment by tvdijen** [comment]
Let me know how v0.9.7 works out for you

**Comment by klausdk** [comment]
Great! It seems to work. "Knackebrøt" returns "Knackebrøt" !

I'm not on the corporate network until tomorrow, but I think it should work.

**Comment by klausdk** [comment]
Oops.. It generates warnings : 

```
preg_match(): Compilation failed: range out of order in character class at offset 16 in /var/www/simplesamlphp/modules/ldap/lib
/Auth/Ldap.php on line 867
```

Below should fix it : 

```
-|| preg_match('~[^\x20-\x7E\x80-xFF\t\r\n]~', $value) > 0
+|| preg_match('~[^\x20-\x7E\x80-\xFF\t\r\n]~', $value) > 0
```

**Comment by tvdijen** [comment]
😭 This change keeps haunting me

**Comment by tvdijen** [comment]
Fixed in 0.9.8
